client/src/js/services/receipts/ReceiptModal.js

```diff
@@ -0,0 +1,55 @@
+angular.module('bhima.services')
+.service('ReceiptModal', ReceiptModal);
+
+ReceiptModal.$inject = ['$uibModal', 'ReceiptService'];
+  
+/** 
+ * Receipts Modal Service 
+ *
+ * This service is responsible for combining receipt service data with the 
+ * receipts modal controller and providing a clean interface to be used within 
+ * module controllers. 
+ *  
+ * @todo Discuss how the render target PDF should be templated. Suggestion: 
+ *       - IF pdf   - use /src/services/receipts/modal/pdf.tmpl.html
+ *       - IF json  - specifiy individual target /partials/patient_invoice/receipt/...
+ *
+ * @module services/receipts/ReceiptModal
+ */
+function ReceiptModal(Modal, Receipts) { 
+  var service = this;
+  
+  var modalConfiguration = { 
+    templateUrl : '/js/services/receipts/modal/receiptModal.tmpl.html',
+    controller  : 'ReceiptModalController as ReceiptCtrl',
+    size        : 'md',
+    backdrop    : 'static',
+    animation   : false
+  };
+  
+  // expose available receipts
+  service.invoice = invoice;
+
+  function invoice(uuid) { 
+
+    /** @todo Discuss if these should be overridable from the controller or if the config should be set here */
+    var options = { 
```

Comment:
@sfount, that is much clearer and makes a lot more sense.  Thanks for the explanation!

I can't think of a cleaner implementation.  LGTM.